Data/DALBranch.cs

```diff
@@ -3,39 +3,84 @@
 using System.Linq;
 using System.Threading.Tasks;
 using DiAnterExpress.Models;
+using Microsoft.EntityFrameworkCore;
 
 namespace DiAnterExpress.Data
 {
     public class DALBranch : IBranch
     {
-        public DALBranch()
+        private ApplicationDbContext _db;
+        public DALBranch(ApplicationDbContext db)
         {
-
+            _db = db;
         }
 
         public Task<Branch> Delete(int id)
         {
             throw new NotImplementedException();
         }
 
-        public Task<IEnumerable<Branch>> GetAll()
+        public async Task DeleteById (int id)
         {
-            throw new NotImplementedException();
+            var result = await GetById(id);
+            if (result == null) throw new Exception("Data tidak ditemukan!");
+            try
+            {
+                _db.Branches.Remove(result);
+                await _db.SaveChangesAsync();
+            }
+            catch (DbUpdateException dbEx)
+            {
+                throw new Exception($"Error: {dbEx.Message}");
+            }
         }
 
-        public Task<Branch> GetById(int id)
+        public async Task<IEnumerable<Branch>> GetAll()
         {
-            throw new NotImplementedException();
+            var results = await _db.Branches.OrderBy(b => b.Name).ToListAsync();
+            return results;
         }
 
-        public Task<Branch> Insert(Branch obj)
+        public async Task<Branch> GetById(int id)
         {
-            throw new NotImplementedException();
+            var result = await _db.Branches.Where(s => s.Id == Convert.ToInt32(id)).SingleOrDefaultAsync<Branch>();
```

Comment:
Ini sebenarnya nggak perlu `Convert.ToInt32(id)` lagi, karena `id` ini udah int
```suggestion
            var result = await _db.Branches.Where(s => s.Id == id).SingleOrDefaultAsync();
```

Reply:
Oke akan di fix

Reply:
Oke sudah di fix